main.js

```diff
@@ -1,60 +1,59 @@
-  {text: 'Toto, I\’ve got a feeling we\’re not in Kansas anymore.', movie: 'The Wizard of Oz'},
-  {text: 'Here\’s looking at you, kid.', movie: 'Casablanca'},
-  {text: 'As God is my witness, I\’ll never be hungry again.', movie: 'Gone with the Wind'},
-  {text: 'Say \‘hello\’ to my little friend!', movie: 'Scarface'},
-  {text: 'I feel the need — the need for speed!', movie: 'Top Gun'},
-  {text: 'Houston, we have a problem.', movie: 'Apollo 13'},
-  {text: 'Nobody puts Baby in a corner.', movie: 'Dirty Dancing'},
-  {text: 'I\’ll get you, my pretty, and your little dog, too!', movie: 'The Wizard of Oz'},
-  {text: 'One morning I shot an elephant in my pajamas. How he got in my pajamas, I don\’t know.', movie: 'Animal Crackers'},
-  {text: 'You\’ve got to ask yourself one question: \‘Do I feel lucky?\’ Well, do ya, punk?', movie: 'Dirty Harry'},
-  {text: 'Hold on to your butts.', movie: 'Jurassic Park'},
-  {text: 'As far back as I can remember, I always wanted to be a gangster.', movie: 'Goodfellas'},
-  {text: 'Human sacrifice! Dogs and cats living together. Mass hysteria!', movie: 'Ghostbusters'}
-];
-const displayArea = document.getElementById('display-area');
-
-bindEvents();
-displayNewQuote();
-
-function bindEvents() {
-  const newQuoteButton = document.getElementById('new-quote-button');
-  newQuoteButton.addEventListener('click', handleNewQuoteClick);
-}
-
-function displayNewQuote() {
-  const quotePara = document.getElementById('quote');
-  const refCite = document.getElementById('ref');
-  const quote = getRandomQuote(quotes);
-
-  quotePara.innerText = '\“' + quote['text'] + '\”';
-  refCite.innerText = quote['movie'];
-  displayArea.classList.toggle('show');
-  updateTwitterIntent(quote);
-};
-
-function getRandomIndex(array) {
-  return Math.floor(Math.random() * array.length);
-}
-
-function getRandomQuote(quotes) {
-  const index = getRandomIndex(quotes);
-  return quotes[index];
-}
-
-function handleNewQuoteClick() {
-  displayArea.classList.toggle('show');
-  setTimeout(displayNewQuote, 350);
-}
-
-function updateTwitterIntent(quote) {
-  const twitterLink = document.getElementById('tweet-quote-link');
-  const intentAddress = `https://twitter.com/intent/tweet?text=“${quote.text}”  — ${quote.movie}`;
-  twitterLink.setAttribute('href', intentAddress);
+fetch('quotes.json')
+  .then(response => {
+    if (!response.ok) {
+      document.getElementById('quote-box').textContent = `
+        Network request failed with status 
+        ${response.status}: ${response.statusText}. 
+        Please refresh to try again.
+      `;
+    }
+    return response.json();
+  })
+  .then(json => {
+    const quotes = json;
+    init(quotes);
+  });
+
+function init(quotes) {
+  const displayArea = document.getElementById('display-area');
+  bindEvents();
+  displayNewQuote();
+
+  function bindEvents() {
```

Comment:
all the subsequent functions are within the init function. is there a reason for it? do you think this is semantically correct?

Reply:
I did this so that when the `quotes` argument is passed to `init()`, it is scoped properly for the other functions to access it. As to whether it is semantically correct -- I don't know. Much of what is going on inside of `init()` doesn't have anything to do with initializing the app, so maybe that would be an argument against taking this approach.

I suppose I could just as easily declare a global variable `quotes` at the very top and bind it to the `json` response object at line 13 without passing it to `init()`.